### DISCUSSION TITLE
Probe doesn't extend when Z auto homing

### DISCUSSION THREAD

**Discussion by hahmlet** [Jyers/Marlin#1262] [Q&A] [answered]
Hello,

I have a Ender 3V2 4.2.2 board with BLTouch (no adapter board) and installed both v1.3.5 and [v1.2.2 ](https://github.com/Jyers/Marlin/releases/tag/v1.2.2) and got the same issue:

Auto home will home X and Y properly, but when homing Z, the Z axis moves up and down, the BLTouch just never extends the probe and so the nozzle smashes into the glass bed. I've made no changes to the firmware install other than set X and Y offset.

When the machine boots on, it extends the probe a few times so I know it can extend it, it just doesn't during Z axis homing.

I have double and triple checked my wiring. Servo wiring is correct and BLTouch is wired to the five pin connector. Brown, orange, yellow, black, and while starting from the bottom up. The connector on the BLTouch end is the same.

**Comment by Amt234** [comment]
Could you send the link to the website you got it from?

**Comment by hahmlet** [comment]
https://www.amazon.com/dp/B076PQG1FF/ref=cm_sw_r_cp_apap_6tBIUtuWwTaMp

**Reply by Amt234** [reply]
I have heard that sometimes Marlin firmware doesn't work with Bl-touches, but creality firmware does.

**Reply by Amt234** [reply]
I got this from the creality website

https://drive.google.com/file/d/1q_7urc1GFNcxzH38XOTUs2ZV0ul8AgU_/view?usp=sharing

**Reply by Amt234** [reply]
Also what slicer do you use?

**Reply by Amt234** [reply]
![Screenshot (3)](https://user-images.githubusercontent.com/88695784/129132216-080331df-71eb-4668-9acc-89850f552102.png)
(Only if you use cura) Add in this code under G28,                 Make sure it looks just like this

G29 ; Auto bed-level (BL-Touch)

**Reply by Amt234** [reply]
I used the ender 3 pro firmware (not ender 3) and then just named it Ender 3 V2.

Please click on the image to see what it actually says

**Comment by hahmlet** [comment]
I have the Ender 3 V2 and Prusa Slicer. I've tried that firmware and no
improvement. Do you have a code recommendation for my Gcode for Prusa?

**Comment by Jyers** [comment, marked as answer]
Honestly sounds like a hardware issue if anything, maybe a faulty probe?
